rpc: TestRemoteOffsetUnhealthy failed

rpc.TestRemoteOffsetUnhealthy [failed](https://tanzanite.cluster.engflow.com/invocations/default/4c6222b7-5ab5-496c-a458-5f8a2f50e55a?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvcnBjOnJwY190ZXN0) on release-24.1.2-rc @ [4037be1494ff8f3873edfa2510a7fa79545b413e](https://github.com/cockroachdb/cockroach/commits/4037be1494ff8f3873edfa2510a7fa79545b413e):


```
E240624 09:40:36.486099 383 2@rpc/peer.go:598  [rnode=2,raddr=127.0.0.1:40621,class=default,rpc] 237  failed connection attempt (last connected 0s ago): operation "conn heartbeat" timed out after 169ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
E240624 09:40:36.488023 443 2@rpc/peer.go:598  [rnode=3,raddr=127.0.0.1:44903,class=default,rpc] 238  failed connection attempt (last connected 0s ago): operation "conn heartbeat" timed out after 189ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
E240624 09:40:36.490153 459 2@rpc/peer.go:598  [rnode=4,raddr=127.0.0.1:41117,class=default,rpc] 239  failed connection attempt (last connected 0s ago): operation "conn heartbeat" timed out after 161ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
E240624 09:40:36.494840 398 2@rpc/peer.go:598  [rnode=3,raddr=127.0.0.1:44903,class=default,rpc] 240  failed connection attempt (last connected 0s ago): operation "conn heartbeat" timed out after 190ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:37.042554 374 testutils/soon.go:42  [-] 241  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 103ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:38.048636 374 testutils/soon.go:42  [-] 242  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 132ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:39.079836 374 testutils/soon.go:42  [-] 243  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 142ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:40.087498 374 testutils/soon.go:42  [-] 244  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 150ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:41.090538 374 testutils/soon.go:42  [-] 245  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 103ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:42.122224 374 testutils/soon.go:42  [-] 246  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 118ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:43.132358 374 testutils/soon.go:42  [-] 247  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 100ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:44.150979 374 testutils/soon.go:42  [-] 248  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 116ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:45.157326 374 testutils/soon.go:42  [-] 249  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 133ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:46.170898 374 testutils/soon.go:42  [-] 250  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 154ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:47.187568 374 testutils/soon.go:42  [-] 251  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 159ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:48.193209 374 testutils/soon.go:42  [-] 252  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 107ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:49.194960 374 testutils/soon.go:42  [-] 253  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 109ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:50.196663 374 testutils/soon.go:42  [-] 254  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 103ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:51.202570 374 testutils/soon.go:42  [-] 255  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 166ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:52.207775 374 testutils/soon.go:42  [-] 256  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 111ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:53.212373 374 testutils/soon.go:42  [-] 257  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 155ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:54.223202 374 testutils/soon.go:42  [-] 258  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 158ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:55.268041 374 testutils/soon.go:42  [-] 259  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 106ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:56.289693 374 testutils/soon.go:42  [-] 260  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 118ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:57.297187 374 testutils/soon.go:42  [-] 261  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 123ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:58.309987 374 testutils/soon.go:42  [-] 262  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 132ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:40:59.314914 374 testutils/soon.go:42  [-] 263  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 100ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:00.317050 374 testutils/soon.go:42  [-] 264  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 163ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:01.326062 374 testutils/soon.go:42  [-] 265  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 102ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:02.364066 374 testutils/soon.go:42  [-] 266  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 158ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:03.367777 374 testutils/soon.go:42  [-] 267  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 123ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:04.383566 374 testutils/soon.go:42  [-] 268  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 114ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:05.393391 374 testutils/soon.go:42  [-] 269  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 129ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:06.400405 374 testutils/soon.go:42  [-] 270  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 145ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:07.461079 374 testutils/soon.go:42  [-] 271  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 130ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:08.466399 374 testutils/soon.go:42  [-] 272  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 167ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:09.539574 374 testutils/soon.go:42  [-] 273  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 116ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:10.540531 374 testutils/soon.go:42  [-] 274  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 157ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:11.578665 374 testutils/soon.go:42  [-] 275  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 101ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:12.600007 374 testutils/soon.go:42  [-] 276  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 104ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:13.620232 374 testutils/soon.go:42  [-] 277  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 144ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:14.621297 374 testutils/soon.go:42  [-] 278  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 132ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:15.627877 374 testutils/soon.go:42  [-] 279  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 108ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:16.630023 374 testutils/soon.go:42  [-] 280  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 106ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:17.669041 374 testutils/soon.go:42  [-] 281  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 116ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:18.721521 374 testutils/soon.go:42  [-] 282  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 147ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:19.722409 374 testutils/soon.go:42  [-] 283  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 161ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
I240624 09:41:20.725577 374 testutils/soon.go:42  [-] 284  SucceedsSoon: initial connection heartbeat failed: operation "conn heartbeat" timed out after 151ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
    context_test.go:1166: condition failed to evaluate within 3m45s: initial connection heartbeat failed: operation "conn heartbeat" timed out after 151ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
--- FAIL: TestRemoteOffsetUnhealthy (226.19s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRemoteOffsetUnhealthy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39770

--- a/pkg/rpc/BUILD.bazel
+++ b/pkg/rpc/BUILD.bazel
@@ -120,6 +120,10 @@ go_test(
     ],
     data = glob(["testdata/**"]),
     embed = [":rpc"],
+    exec_properties = select({
+        "//build/toolchains:is_heavy": {"test.Pool": "large"},
+        "//conditions:default": {"test.Pool": "default"},
+    }),
     deps = [
         "//pkg/base",
         "//pkg/clusterversion",

